src/Utility/InputValidator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+
+namespace WpOrg\Requests\Utility;
+
+use ArrayAccess;
+use CurlHandle;
+use Traversable;
+
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+final class InputValidator {
+
+	/**
+	 * Verify that a received input parameter is of type string or is "stringable".
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_string_or_stringable($input) {
+		return is_string($input) || self::is_stringable_object($input);
+	}
+
+	/**
+	 * Verify whether a received input parameter is usable as an integer array key.
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_numeric_array_key($input) {
+		if (is_int($input)) {
+			return true;
+		}
+
+		if (!is_string($input)) {
+			return false;
+		}
+
+		return (bool) preg_match('`^-?[0-9]+$`', $input);
+	}
+
+	/**
+	 * Verify whether a received input parameter is "stringable".
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_stringable_object($input) {
+		return is_object($input) && method_exists($input, '__toString');
+	}
+
+	/**
+	 * Verify whether a received input parameter is _accessible as if it were an array_.
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_accessible_as_array($input) {
```

Comment:
Potential alternative names:
- `is_indexable()`
- `has_array_access()`

Reply:
Updated in all relevant place to `has_array_access()`.